packages/camera/camera_android_camerax/lib/src/android_camera_camerax.dart

```diff
@@ -509,6 +564,51 @@ class AndroidCameraCameraX extends CameraPlatform {
     return ImageFormatGroup.unknown;
   }
 
+  // Methods concerning camera state:
+
+  /// Adds observers to the [LiveData] of the [CameraState] of the current
+  /// [camera], saved as [liveCameraState].
+  ///
+  /// If a previous [liveCameraState] was stored, existing observers are
+  /// removed, as well.
+  Future<void> _updateLiveCameraState(int cameraId) async {
+    final CameraInfo cameraInfo = await camera!.getCameraInfo();
```

Comment:
I think this gets at the bigger question as to why `cameraId` is used throughout the Flutter camera API to begin with. Really, `cameraId` is the ID of the Flutter surface texture used to render a preview to, and thus, is almost totally unrelated to the actual `camera` instance we use to do any camera functionality. The Flutter surface texture ID is returned by calls to `createCamera(...)`, so I assume it is just an ID that folks using the plugin can mostly use to filter through `CameraEvents` added to the various camera event streams (because they are constructed with an ID) because otherwise, at least in the Android camera implementations, it goes mostly unused.

I'm on the fence about renaming `cameraId`, because while it would be more useful to name it something more like `flutterSurfaceTextureId`, it's not named that in the actual camera platform interface, so it could cause more confusion for folks trying to contribute to the plugin.

However, I am on board for bettering documentation in the implementation around what it is (always referring to the Flutter surface texture ID/`createCamera`), when it is actually being used in a method, and if it is being used, exactly how. I'll file an issue before landing this for that.

Reply:
https://github.com/flutter/flutter/issues/126270 filed for improving documentation